configuration/src/main/java/io/camunda/configuration/UnifiedConfigurationHelper.java

```diff
@@ -0,0 +1,253 @@
+    }
+
+    // Backwards compatibility supported  only if values match
+    if (legacyConfigAllowed && onlyIfValuesMatch) {
+      final T legacyValue = getLegacyValue(legacyProperties, expectedType);
+      return backwardsCompatibilitySupportedOnlyIfValuesMatch(
+          legacyProperties, legacyValue, newProperty, newValue);
+    }
+
+    // Backwards compatibility supported
+    final T legacyValue = getLegacyValue(legacyProperties, expectedType);
+    return backwardsCompatibilitySupported(legacyProperties, legacyValue, newProperty, newValue);
+  }
+
+  private static <T> T getLegacyValue(
+      final Set<String> legacyProperties, final Class<T> expectedType) {
+    final Set<T> legacyValues = new HashSet<>();
+
+    for (final String legacyProperty : legacyProperties) {
+      final String strValue = environment.getProperty(legacyProperty);
+      final T legacyValue = parseLegacyValue(strValue, expectedType);
+      legacyValues.add(legacyValue);
+    }
+
+    if (legacyValues.size() > 1) {
+      throw new UnifiedConfigurationException(
+          String.format(
+              "Ambiguous legacy configuration. Legacy properties: %s; Legacy values: %s",
+              String.join(", ", legacyProperties), legacyValues.toString()));
+    }
+
+    return legacyValues.iterator().next();
+  }
+
+  private static <T> T backwardsCompatibilityNotSupported(
+      final Set<String> legacyProperties, final String newProperty, final T newValue) {
+    final boolean legacyPresent = legacyConfigPresent(legacyProperties);
+    final boolean newPresent = newConfigPresent(newProperty);
+
+    if (!legacyPresent && !newPresent) {
+      // Legacy config: not present
+      // New config...: not present
+      // We can return the default value
+      return newValue;
+    } else if (!legacyPresent && newPresent) {
+      // Legacy config: not present
+      // New config...: present
+      // We can return newValue, that's either the default or the configured one
+      return newValue;
+    } else if (legacyPresent && !newPresent) {
+      // Legacy config: present
+      // New config...: not present
+      // The legacy configuration is no longer allowed -> error
+      final String errorMessage =
+          String.format(
+              "The following legacy configuration properties are no longer supported and must be removed in favor of '%s': %s",
+              newProperty, String.join(", ", legacyProperties));
+      throw new UnifiedConfigurationException(errorMessage);
+    } else {
+      // Legacy config: present
+      // New config...: present
+      // The legacy config should be removed entirely -> error
+      final String errorMessage =
+          String.format(
+              "The following legacy configuration properties are no longer supported and must be removed: %s",
+              String.join(", ", legacyProperties));
+      throw new UnifiedConfigurationException(errorMessage);
```

Comment:
:x: If new is present, then we don't want to throw exception. But just log a warning that the legacy property should be removed.